src/test-run/index.js

```diff
@@ -121,6 +121,10 @@ export default class TestRun extends EventEmitter {
         this.requestHooks = Array.from(this.test.requestHooks);
 
         this._initRequestHooks();
+
+        this.browserConnection.once('disconnected', err => this.disconnect(err));
+        this.browserConnection.once('opened', () => this.emit('stop-waiting'));
```

Comment:
To restart test we need to execute TestDone command, so we wait the moment when browser is opened. Othervise in case when hanged browser has alived suddenly the TestDone command can be enqueued there.
Moreover we are waiting for opening because of the mechanism of closing browsers. If the test will be rejected immediately then new browser window will be also opened immediatelly. But we need to give some time for browser-tools to get browser window handler.